torch/csrc/jit/codegen/cuda/scheduler/vectorize_helper.cpp

```diff
@@ -17,171 +18,881 @@ namespace fuser {
+          if (inner_pos > 0) {
+            ids.erase(ids.begin(), ids.begin() + in_pos);
+          }
+        }
+      } else {
+        // Only inner matches, mark a partial match
+        ids[inner_pos] = split->in();
+        // Mark the partial match, inner could already be a partial match
+        partial_projected_extent_[split->in()] =
+            getMaybePartialMappedExtent(split->inner());
+        // Clear to the left of split in since it partially maps
+        ids.erase(ids.begin(), ids.begin() + inner_pos);
+      }
+    } else if (const Merge* merge = dynamic_cast<const Merge*>(expr)) {
+      auto find_out_it = std::find(ids.begin(), ids.end(), merge->out());
+      if (find_out_it == ids.end()) {
+        continue;
+      }
+
+      auto out_pos = std::distance(ids.begin(), find_out_it);
+
+      if (!hasPartialExtent(merge->out())) {
+        // No partial map in output, so simply map through to inputs.
+        ids[out_pos] = merge->outer();
+        ids.insert(ids.begin() + out_pos + 1, merge->inner());
+      } else {
+        // This could be done completely symbolically by adding more expressions
+        // into the graph, for example the inner patial extent could be:
+        //   auto inner_partial_extent =
+        //   SimplifyingIrBuilder::minExpr(getMaybePartialMappedExtent(merge->out()),
+        //   merge->inner()->extent());
+        // and outer partial extent:
+        //   auto outer_partial_extent =
+        //   SimplifyingIrBuilder::divExpr(getMaybePartialMappedExtent(merge->out()),
+        //   merge->inner()->extent())
+        //
+        // However, for right now we'll assume no partial outer dimension maps
+        // unless the extent of merge->out and merge->inner are compile time
+        // constant. If not compile time constant we'll take:
+        //   min(merge->inner()->extent(), the partial extent of merge->out())
+        // for the inner partial map, and assume outer doesn't partially map.
+        //
+        // TODO: Improve to do the above. The challenge here is adding correct
+        // conditionals to evaluate (and add conditional evaluation in expr
+        // evaluator) to check vectorization especially like (out_extent >
+        // inner_extent).
+        if (!getMaybePartialMappedExtent(merge->out())->isConstInt() ||
+            !merge->inner()->extent()->isConstInt()) {
+          // We don't know at compile time if the partial extent of merge->out
+          // is bigger than merge->inner or not, so generate an expressions we
+          // can evaluate at runtime for the inner partial mapping extent. Don't
+          // attempt to map merge->outer, just assume it doesn't partially map.
+          ids[out_pos] = merge->inner();
+          partial_projected_extent_[merge->inner()] =
+              SimplifyingIrBuilder::minExpr(
+                  getMaybePartialMappedExtent(merge->out()),
+                  merge->inner()->extent());
```

Comment:
I don't think this is correct. I feel that, for this case, we should just drop both the inner and the outer. For example, if you have
```
15 = merge(2, 3)
```
and
```
getMaybePartialMappedExtent(merge->out()) == 2
```
I don't think we can simply propagate this `2` to inner.

Reply:
Why? Aren't we assuming the merge inner and outer are contiguous?

Reply:
For example, if we have
```
T0[5, 2, 3]
T1[5, 6] = view(T0)
T2[5, 3, 2] = view(T1)
Tref[3, 5, 2] = transpose(T2)
```
For this case, can we vectorize T0 by 2?

Reply:
I think I have a counterexample now. It is slightly modified from
```
T0[5, 2, 3]
T3[5, 2, 3] = set(T0)
T1[5, 6] = view(T3)
T2[5, 3, 2] = view(T1)
Tref[3, 5, 2] = transpose(T2)
```
So `T3` is guaranteed to be contiguous, but `T0` is not. If the contiguity of `T0` is `(false, false, true)`, then we get wrong result.

Reply:
Yeah, I'm sure how it would work in that case. The comment on `ContiguousInnerDimensionMapper` indicates the reference is expected to be contiguous, but not sure about other tensors.